src/Action.hs

```diff
@@ -867,8 +858,12 @@ eventLogUserVotesOnComment ck@(CommentKey _ ideaId parentIds _) v = do
 
 eventLogUserDelegates ::
       (ActionUserHandler m, ActionPersist m, ActionCurrentTimestamp m, ActionLog m)
-      => DScope -> AUID User -> m ()
-eventLogUserDelegates scope delegateId = do
+      => Bool -> DScope -> AUID User -> m ()
+eventLogUserDelegates setOrWithdraw scope delegateId = do
```

Comment:
I would introduce two functions on top of this one. The two would be `eventLogUserSetsDelegate` and `eventLogUserWithdrawsDelegate`.